packages/xo-server/src/api/network.mjs

```diff
@@ -120,3 +120,19 @@ delete_.params = {
 delete_.resolve = {
   network: ['id', 'network', 'administrate'],
 }
+
+// =================================================================
+
+export async function updatePurpose({ networkId, oldPurpose, newPurpose }) {
+  const xapi = this.getXapi(networkId)
+  const networkRef = this.getObject(networkId)._xapiRef
```

Comment:
I think it should be `this.getObject(networkId).$ref`
Also, it can be empty you have to handle it